### PR TITLE
The rest of coding easy blog features

### PR DESCRIPTION
**Pull request by muhammadgalhoum**
# Adding more features to the blog 
- search
- sidebar
- user profile
- reset the password and delete the account from the profile page
- add, update and delete a category 
- only the user who creates the post or category is able to edit or delete it

https://user-images.githubusercontent.com/56798376/166122223-016b8dc1-fa75-4c53-a448-82d8f7bdc4f6.mp4

## ✔️ Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by muhammadgalhoum** [comment]
### Can you please review this PR @singhmansi25 @KKhushhalR2405

**Comment by KKhushhalR2405** [comment]
@muhammadgalhoum Can you please specify which PR is doing what?

**Comment by muhammadgalhoum** [comment]
> @muhammadgalhoum Can you please specify which PR is doing what?

Please read the description in each PR

**Comment by arpit456jain** [comment]
@muhammadgalhoum i have merged ur previous PR can u solve these merge conflicts

**Comment by muhammadgalhoum** [comment]
> @muhammadgalhoum i have merged ur previous PR can u solve these merge conflicts

Solved

File: templates/users/profile.html [hunk 1]
```diff
@@ -9,12 +9,12 @@
           <h2 class="account-heading">{{ user.username }}</h2>
           <p class="text-secondary">{{ user.email }}</p>
           <div>
-            {% comment %} <a class="btn btn-secondary btn-sm mt-1 mb-1" href="{% url 'password_reset' %}">
+            <a class="btn btn-secondary btn-sm mt-1 mb-1" href="{% url 'password_reset' %}">
               <i class="fas fa-edit"></i>&nbsp;Reset Password
```
Comment (arpit456jain) [inline templates/users/profile.html:13]: @muhammadgalhoum in this PR there is nothing to review

**Comment by arpit456jain** [comment]
@muhammadgalhoum in this PR u are just adding the comments nothing else

**Comment by muhammadgalhoum** [comment]
> @muhammadgalhoum in this PR u are just adding the comments nothing else

I did, but as I mentioned before, I worked hard and hard to get these results and instead of deleting all the code for these features and rewriting them again, I just made it a comment so that it wouldn't appear in the UI and then I showed it again and I did it to not write the code again and also to avoid any conflict When making PR due to incomplete code and also so that you do not have any problems when trying your code

Therefore, please appreciate my time, work and effort to obtain these desired results
@arpit456jain

**Comment by arpit456jain** [comment]
@muhammadgalhoum  ok  i have to suggested some small changes creating a issue for that do them i will give lvl3 on that PR

**Comment by muhammadgalhoum** [comment]
> @muhammadgalhoum ok i have to suggested some small changes creating a issue for that do them i will give lvl3 on that PR

Please put at least Level 1 on this PR

**Comment by arpit456jain** [comment]
@muhammadgalhoum its not possible and ethical try to understand check #267

**Pull request merged**